tokio/src/sync/batch_semaphore.rs

```diff
@@ -150,10 +150,9 @@ impl Semaphore {
         waiters.closed = true;
         while let Some(mut waiter) = waiters.queue.pop_back() {
             let waker = unsafe { waiter.as_mut().waker.with_mut(|waker| (*waker).take()) };
-
-            waker
-                .expect("if a node is in the wait list, it must have a waker")
-                .wake();
+            if let Some(waker) = waker {
+                waker.wake();
```

Comment:
I guess we keep waking inside the lock here since it's not considered to be the hot path?
Seems fine to me.